geode-core/src/integrationTest/java/org/apache/geode/distributed/LocatorIntegrationTest.java

```diff
@@ -63,162 +64,168 @@
+        .isEqualTo(GemFireCacheImpl.getInstance().getMyId());
+  }
 
   @Test
   public void testHandlersAreWaitedOn() throws Exception {
-    Properties dsprops = new Properties();
-    int jmxPort = getRandomAvailablePort(SOCKET);
-    dsprops.setProperty(MCAST_PORT, "0");
-    dsprops.setProperty(ENABLE_CLUSTER_CONFIGURATION, "false");
-    dsprops.setProperty(LOCATOR_WAIT_TIME, "1"); // seconds
-    dsprops.setProperty(LOG_FILE, "");
-    locator = Locator.startLocatorAndDS(port, null, dsprops);
+    Properties configProperties = new Properties();
+    configProperties.setProperty(MCAST_PORT, "0");
+    configProperties.setProperty(ENABLE_CLUSTER_CONFIGURATION, "false");
+    configProperties.setProperty(LOCATOR_WAIT_TIME, "1"); // seconds
+    configProperties.setProperty(LOG_FILE, "");
+
+    locator = Locator.startLocatorAndDS(port, null, configProperties);
+
     InternalLocator internalLocator = (InternalLocator) locator;
+
     // the locator should always install a SharedConfigurationStatusRequest handler
-    assertTrue(internalLocator.hasHandlerForClass(SharedConfigurationStatusRequest.class));
+    assertThat(internalLocator.hasHandlerForClass(SharedConfigurationStatusRequest.class)).isTrue();
   }
 
-
   @Test
   public void testBasicInfo() throws Exception {
     locator = Locator.startLocator(port, tmpFile);
-    int boundPort = (port == 0) ? locator.getPort() : port;
+    int boundPort = port == 0 ? locator.getPort() : port;
     TcpClient client = new TcpClient();
     String[] info = client.getInfo(InetAddress.getLocalHost(), boundPort);
-    assertNotNull(info);
-    assertTrue(info.length > 1);
+
+    assertThat(info).isNotNull();
+    assertThat(info.length).isGreaterThanOrEqualTo(1);
   }
 
   @Test
   public void testNoThreadLeftBehind() throws Exception {
-    Properties dsprops = new Properties();
-    dsprops.setProperty(MCAST_PORT, "0");
-    dsprops.setProperty(JMX_MANAGER_START, "false");
-    dsprops.setProperty(ENABLE_CLUSTER_CONFIGURATION, "false");
-
     JGroupsMessenger.THROW_EXCEPTION_ON_START_HOOK = true;
+
+    Properties configProperties = new Properties();
+    configProperties.setProperty(MCAST_PORT, "0");
+    configProperties.setProperty(JMX_MANAGER_START, "false");
+    configProperties.setProperty(ENABLE_CLUSTER_CONFIGURATION, "false");
+
     int threadCount = Thread.activeCount();
-    try {
-      locator = Locator.startLocatorAndDS(port, new File(""), dsprops);
-      locator.stop();
-      fail("expected an exception");
-    } catch (SystemConnectException expected) {
 
-      for (int i = 0; i < 10; i++) {
-        if (threadCount < Thread.activeCount()) {
-          Thread.sleep(1000);
-        }
-      }
+    Throwable thrown = catchThrowable(
+        () -> locator = Locator.startLocatorAndDS(port, new File(""), configProperties));
+
+    assertThat(thrown).isInstanceOf(SystemConnectException.class);
+
+    for (int i = 0; i < 10; i++) {
```

Comment:
why would this not be an awaitility? Is this another tool we've decided to no use. I think this check can be improved using awaitility.

Reply:
The for-loop with sleep and checks is old code. Again, this is not new code. I'm not rewriting the tests -- I'm simply fixing flakiness, fixing IDE warnings, and fixing low-hanging fruit. I'm not planning to rewrite them from scratch.